lib/build_targets/plugins.dart

```diff
@@ -64,151 +64,206 @@ class NativePlugins extends Target {
-APPNAME = flutter_plugins
-type = sharedLib
-profile = $profile-$apiVersion
-
-USER_CPP_DEFS = TIZEN_DEPRECATION DEPRECATION_WARNING FLUTTER_PLUGIN_IMPL
-USER_CPPFLAGS_MISC = -c -fmessage-length=0
-USER_LFLAGS = -Wl,-rpath='\$\$ORIGIN'
-''');
-
     // Check if there's anything to build.
     final List<TizenPlugin> nativePlugins =
         await findTizenPlugins(project, nativeOnly: true);
     if (nativePlugins.isEmpty) {
-      rootDir.deleteSync(recursive: true);
-
       depfileService.writeToFile(
         Depfile(inputs, outputs),
         environment.buildDir.childFile('tizen_plugins.d'),
       );
       return;
     }
 
-    // Prepare for build.
+    // Create a dummy project in the build directory.
+    final Directory rootDir = environment.buildDir
+        .childDirectory('tizen_plugins')
+          ..createSync(recursive: true);
     final Directory includeDir = rootDir.childDirectory('include')
       ..createSync(recursive: true);
     final Directory libDir = rootDir.childDirectory('lib')
       ..createSync(recursive: true);
 
-    final List<String> userIncludes = <String>[];
-    final List<String> userSources = <String>[];
+    final BuildMode buildMode = buildInfo.buildInfo.mode;
+    final String buildConfig = buildMode.isPrecompiled ? 'Release' : 'Debug';
+    final Directory engineDir =
+        getEngineArtifactsDirectory(buildInfo.targetArch, buildMode);
+    final Directory commonDir = engineDir.parent.childDirectory('tizen-common');
+
+    final Directory clientWrapperDir =
+        commonDir.childDirectory('cpp_client_wrapper');
+    final Directory publicDir = commonDir.childDirectory('public');
+    clientWrapperDir
+        .listSync(recursive: true)
+        .whereType<File>()
+        .forEach(inputs.add);
+    publicDir.listSync(recursive: true).whereType<File>().forEach(inputs.add);
+
+    assert(tizenSdk != null);
+    final TizenManifest tizenManifest =
+        TizenManifest.parseFromXml(tizenProject.manifestFile);
+    final String profile = tizenManifest.profile;
+    final String apiVersion = tizenManifest.apiVersion;
+    final Rootstrap rootstrap = tizenSdk.getFlutterRootstrap(
+      profile: profile,
+      apiVersion: apiVersion,
+      arch: buildInfo.targetArch,
+    );
+    inputs.add(tizenProject.manifestFile);
+
     final List<String> userLibs = <String>[];
+    final List<String> pluginClasses = <String>[];
 
     for (final TizenPlugin plugin in nativePlugins) {
+      // Create a copy of the plugin to allow editing its projectFile.
+      final TizenPlugin pluginCopy = plugin.copyWith(
+        directory: environment.fileSystem.systemTempDirectory.createTempSync(),
+      );
+      copyDirectory(plugin.directory, pluginCopy.directory);
+
+      final List<String> properties = <String>[];
+      for (String line in plugin.projectFile.readAsLinesSync()) {
+        if (line.startsWith('type =')) {
+          line = 'type = staticLib';
+        }
+        properties.add(line);
+      }
```

Comment:
maybe this part can be extracted to `_ensureStaticLibType(plugin)`. 
And I the type is not `staticLib`, I think the information that is has been changed should be notified as some log message.

Reply:
Nice idea!